internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -3285,6 +3328,26 @@ func expandKubernetesClusterAutoScalerProfile(input []interface{}) *managedclust
 	}
 }
 
+func expandKubernetesClusterAzureKeyVaultKms(d *pluginsdk.ResourceData, input []interface{}) *managedclusters.AzureKeyVaultKms {
+	if ((input == nil) || len(input) == 0) && d.HasChanges("key_vault_kms") {
+		return &managedclusters.AzureKeyVaultKms{
+			Enabled: utils.Bool(false),
+		}
+	} else if (input == nil) || len(input) == 0 {
+		return nil
+	}
+
+	raw := input[0].(map[string]interface{})
+	kvAccess := managedclusters.KeyVaultNetworkAccessTypes(*utils.String(raw["key_vault_network_access"].(string)))
+
+	return &managedclusters.AzureKeyVaultKms{
+		Enabled:               utils.Bool(raw["enabled"].(bool)),
+		KeyId:                 utils.String(raw["key_id"].(string)),
+		KeyVaultNetworkAccess: &kvAccess,
+		KeyVaultResourceId:    utils.String(raw["key_vault_resource_id"].(string)),
```

Comment:
Would `raw["key_vault_resource_id"]` here be `nil`?

Reply:
Hi @mkilchhofer , since the `key_vault_resource_id` is an optional argument, would `raw["key_vault_resource_id"]` return a `nil` and cause panic here?

Reply:
Will try to refactor this and remove it. (see comment of @stephybun https://github.com/hashicorp/terraform-provider-azurerm/pull/19893#discussion_r1064413881)